internal/providers/detect.go

```diff
@@ -160,6 +160,7 @@ func autodetectedRootToProviders(projectContext *config.ProjectContext, rootPath
 					options,
 					hcl.OptionWithTFVarsPaths(append(autoVarFiles.ToPaths(), env.TerraformVarFiles.ToPaths()...), true),
 					hcl.OptionWithModuleSuffix(env.Name),
+					hcl.OptionWithProjectEnv(rootPath.DetectedPath, env.Name),
```

Comment:
I should probably just include this option into `OptionWithModuleSuffix` as the order matters now.

Reply:
agreed, this logic should go into the module suffix option